include/zephyr/drivers/pwm.h

```diff
@@ -484,14 +484,13 @@ static inline int z_impl_pwm_set_cycles(const struct device *dev,
 					uint32_t channel, uint32_t period,
 					uint32_t pulse, pwm_flags_t flags)
 {
-	const struct pwm_driver_api *api =
-		(const struct pwm_driver_api *)dev->api;
+	__ASSERT_NO_MSG(DEVICE_API_IS(pwm, dev));
```

Comment:
As per discussions on Discord (see https://discord.com/channels/720317445772017664/1310670722401833030) the asserts needs to be removed in order not to break out-of-tree drivers.